drivers/dht/dht.c

```diff
@@ -94,106 +113,42 @@ static void dht_read_data(gpio_t dev, uint32_t *data, uint8_t *checksum)
      * [humidity][temperature][checksum]
      */
 
-    /* read all the bits */
-    uint64_t les_bits = 0x00000000000000;
-    for (int c = 0; c < 40; c++) {
-        les_bits <<= 1; /* this is a nop in the first iteration, but
-                           we must not shift the last bit */
-        /* wait for start of bit */
-        while (!gpio_read(dev)) ;
-        unsigned long start = xtimer_now();
-        /* wait for end of bit */
-        while (gpio_read(dev)) ;
-        /* calculate bit length (long 1, short 0) */
-        unsigned long stop = xtimer_now();
-        /* compensate for overflow if needed */
-        if (stop < start) {
-            stop = UINT32_MAX - stop;
-            start = 0;
-        }
-        if ((stop - start) > 40) {
-            /* read 1, set bit */
-            les_bits |= 0x0000000000000001;
-        }
-        else {
-            /* read 0, don't set bit */
-        }
-    }
-
-    *checksum = les_bits & 0x00000000000000FF;
-    *data = (les_bits >> 8) & 0x00000000FFFFFFFF;
-
-    gpio_init(dev, GPIO_DIR_OUT, GPIO_PULLUP);
-    gpio_set(dev);
-}
+    /* read the humidity, temperature, and checksum bits */
+    raw_hum = read(dev->pin, 16);
+    raw_temp = read(dev->pin, 16);
+    csum = (uint8_t)read(dev->pin, 8);
 
-/***********************************************************************
- * public API implementation
- **********************************************************************/
+    /* set pin high again - so we can trigger the next reading by pulling it low
+     * again */
+    gpio_init(dev->pin, GPIO_DIR_OUT, dev->pull);
```

Comment:
yes I do... the pin needs be confiured as output again... l106 configured it as input...